base/compiler/ssair/passes.jl

```diff
@@ -513,7 +514,11 @@ function lift_comparison!(compact::IncrementalCompact,
     # Let's check if we evaluate the comparison for each one of the leaves
     lifted_leaves = nothing
     for leaf in leaves
-        r = egal_tfunc(argextype(leaf, compact), cmp)
+        if is_known_call(stmt, (===), compact)
```

Comment:
this check can be hoisted out of the loop, or maybe even out of the function, since we already check this below. (i.e. maybe just pass in the tfunc to evaluate).